### PR TITLE
[DOC]: Updated references to BibTex format

### LINKED ISSUE #2837: References are formatted properly
## Expected behavior ##

References are formatted properly

## Actual behavior ##

Mark*down* markup != reStructuredText

Screen Shot 2020-07-11 at 6 41 32 am

## Code to reproduce the behavior ##

https://github.com/MDAnalysis/mdanalysis/blob/3e56b70da3b48a523924308e1149d46fadddbb5d/package/MDAnalysis/analysis/contacts.py#L257

## Current version of MDAnalysis ##

2.0.0-dev0 docs

## More ##

It might be easier to just use bibtex and autoformat the references, as the user guide does. Then we'll  also have a consistent style.

### LINKED ISSUE #3766: Docs: Missing references in BibTex format in reference page.
## Expected behavior ##
In docs, the [references](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/doc/sphinx/source/documentation_pages/references.rst)  should display citations following the BibTex format. 

Note: This change was left on purpose given that PR #3663 was getting too large. (See [comment](https://github.com/MDAnalysis/mdanalysis/pull/3663#issuecomment-1127496809)) 

## Actual behavior ##
image

To fix this issue:

- [ ] Update `package/doc/sphinx/source/documentation_pages/references.rst` to citations with BibTex format. 
- [ ] Update [requirements](https://github.com/MDAnalysis/mdanalysis/blame/develop/package/requirements.txt) to sphinx (with no pinned version, correct @IAlibay? See [comment](https://github.com/MDAnalysis/mdanalysis/pull/3663#pullrequestreview-973081862))

### PR DESCRIPTION
**Pull request by ojeda-e**
Fixes #3766 and entirely fixes #2837 

Changes made in this Pull Request:
 - Updated all references in `references.rst` to BibTex format.
 - Added references to `references.bib`.
 - Fixed typos.
 - Fixed links.

PR Checklist
------------
 - N/A Tests
 - [x] Docs
 - N/A CHANGELOG updated
 - [x] Issue raised/referenced

Comments:
This time I added changes split into different small commits, which hopefully will help reviewers to identify the changes in each section of the `references.rst`.
I will leave some specific comments in the code to double-check check I didn't miss anything and that we are not losing information.

### CODE REVIEW

File: package/doc/sphinx/source/documentation_pages/references.rst [hunk 1]
```diff
@@ -65,170 +51,182 @@ Citations for included algorithms and modules
...
+:meth:`~MDAnalysis.core.group.SegmentGroup.shape_parameter`,
+or if you calculate asphericities using
 :meth:`~MDAnalysis.core.group.AtomGroup.asphericity`,
 :meth:`~MDAnalysis.core.group.ResidueGroup.asphericity`,
-:meth:`~MDAnalysis.core.group.SegmentGroup.asphericity`
-please cite [Dima2004b]_.
+:meth:`~MDAnalysis.core.group.SegmentGroup.asphericity`,
+please cite :cite:p:`a-Dima2004`.
 
-.. [Dima2004b] Dima, R. I., & Thirumalai, D. (2004). Asymmetry
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/references.rst:201]: The reference with key `[Dima2004a]` seems to be the same as the one here referred to as `[Dima2004b]`. I left one single reference with key `Dima2004` and grouped all the methods listed above and below.

File: package/doc/sphinx/source/documentation_pages/references.rst [hunk 2]
```diff
@@ -65,170 +51,182 @@ Citations for included algorithms and modules
...
+   :style: MDA
+   :keyprefix: a-
+   :labelprefix: ᵃ
 
-.. [Bansal2000b] Bansal M, Kumar S, Velavan R. HELANAL — A program to
-   characterise helix geometry in proteins. *J. Biomol. Struct. Dyn.* **17**
-   (2000), 811–819
+   Bansal2000
 
 .. _HELANAL: http://www.ccrnp.ncifcrf.gov/users/kumarsan/HELANAL/helanal.html
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/references.rst]: Although I could have added this URL to the bibtex reference, I left it just in case this is preferred. Otherwise, I can pass the link to the reference and show the URL within the reference.

File: package/doc/sphinx/source/documentation_pages/references.rst [hunk 3]
```diff
@@ -65,170 +51,182 @@ Citations for included algorithms and modules
...
-   On the interaction of hyaluronic acid with synovial fluid lipid membranes. *PCCP*
-   **21** (2019), 9845-9857. doi:  `10.1039/C9CP01532A`_
+.. bibliography::
+   :filter: False
+   :style: MDA
+   :keyprefix: a-
+   :labelprefix: ᵃ
 
-.. _`10.1039/C9CP01532A`: http://dx.doi.org/10.1039/C9CP01532A
+   Smith2019
```
Comment (ojeda-e) [inline package/doc/sphinx/source/documentation_pages/references.rst:146]: This reference appears on the `references.rst` page but not in the docstrings of `MDAnalysis.analysis.hydrogenbonds.hbond_analysis`. Please let me know if the preference is to add it to the analysis file as well.

File: package/doc/sphinx/source/documentation_pages/references.rst (hunk 2 above)
Comment (orbeckst) [reply package/doc/sphinx/source/documentation_pages/references.rst]: Leave it separate, the link is to the code, not the paper.

File: package/doc/sphinx/source/documentation_pages/references.rst (hunk 3 above)
Comment (orbeckst) [reply package/doc/sphinx/source/documentation_pages/references.rst:146]: Yes, I think it should be in the analysis file, too — @p-j-smith ?

File: package/MDAnalysis/analysis/helix_analysis.py [hunk 4]
```diff
@@ -33,23 +33,19 @@
...
-([Bansal2000]_ , [Sugeta1967]_ ).
+:cite:p:`Bansal2000` ([Sugeta1967]_ ).
 
 HELANAL_ quantifies the geometry of helices in proteins on the basis of their
 Cα atoms. It can determine local structural features such as the local
 helical twist and rise, virtual torsion angle, local helix origins and
 bending angles between successive local helix axes.
 
-.. _HELANAL: https://pubmed.ncbi.nlm.nih.gov/10798526/
+.. _HELANAL: http://www.ccrnp.ncifcrf.gov/users/kumarsan/HELANAL/helanal.html
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/helix_analysis.py]: This URL is actually broken with a 404

We could link to a page on the wayback machine https://web.archive.org/web/20100818185943/http://www.ccrnp.ncifcrf.gov/users/kumarsan/HELANAL/helanal.html instead — although for right now, I am also fine with just leaving the historic broken link and then worry about link clean-up another time, especially as we don't really have a policy for what to do with broken links.

File: package/MDAnalysis/analysis/helix_analysis.py [hunk 5]
```diff
@@ -33,23 +33,19 @@
 
 This module contains code to analyse protein helices using the
 HELANAL_ algorithm
-([Bansal2000]_ , [Sugeta1967]_ ).
+:cite:p:`Bansal2000` ([Sugeta1967]_ ).
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/helix_analysis.py]: Can you turn Sugeta1967 also into a bib link? It seems inconsistent to mix the two.

File: package/MDAnalysis/analysis/helix_analysis.py [hunk 6]
```diff
@@ -33,23 +33,19 @@
...
 
 HELANAL_ quantifies the geometry of helices in proteins on the basis of their
 Cα atoms. It can determine local structural features such as the local
 helical twist and rise, virtual torsion angle, local helix origins and
 bending angles between successive local helix axes.
 
-.. _HELANAL: https://pubmed.ncbi.nlm.nih.gov/10798526/
+.. _HELANAL: http://www.ccrnp.ncifcrf.gov/users/kumarsan/HELANAL/helanal.html
 
 .. [Sugeta1967] Sugeta, H. and Miyazawa, T. 1967. General method for
```
Comment (orbeckst) [inline package/MDAnalysis/analysis/helix_analysis.py]: Should also be in bib.

File: package/doc/sphinx/source/documentation_pages/references.rst [hunk 7]
```diff
@@ -36,26 +36,12 @@ description of the library. We will publish a complete self-contained
 paper with the upcoming 1.0 release of MDAnalysis, which will then
```
Comment (orbeckst) [inline package/doc/sphinx/source/documentation_pages/references.rst:36]: ... er, yes, well, ... I suppose we _really_ should

File: package/doc/sphinx/source/references.bib [hunk 8]
```diff
@@ -734,4 +734,151 @@ @incollection{Waveren2005
...
+	doi = "10.1039/C9CP01532A"}
+
+@article{Liu2010,
+	author = {Liu, Pu and Agrafiotis, Dimitris K. and Theobald, Douglas L.},
+	title = {Fast determination of the optimal rotational matrix for macromolecular superpositions},
+	journal = {Journal of Computational Chemistry},
+	volume = {31},
+	number = {7},
+	pages = {1561-1563},
+	doi = {https://doi.org/10.1002/jcc.21439},
```
Comment (orbeckst) [inline package/doc/sphinx/source/references.bib]: Malke doi just the id, not the URL

File: package/doc/sphinx/source/references.bib [hunk 9]
```diff
@@ -734,4 +734,151 @@ @incollection{Waveren2005
...
+	title = {HELANAL: A Program to Characterize Helix Geometry in Proteins},
+	journal = {Journal of Biomolecular Structure and Dynamics},
+	volume = {17},
+	number = {5},
+	pages = {811-819},
+	year  = {2000},
+	publisher = {Taylor & Francis},
+	doi = {10.1080/07391102.2000.10506570},
+	note ={PMID: 10798526},
+	url = {http://www.ccrnp.ncifcrf.gov/users/kumarsan/HELANAL/helanal.html}
```
Comment (orbeckst) [inline package/doc/sphinx/source/references.bib]: remove the URL of the software (the link is broken anyway)

File: package/doc/sphinx/source/references.bib [hunk 10]
```diff
@@ -734,4 +734,151 @@ @incollection{Waveren2005
...
+
+@article{Leo-Macias2005,
+	title = {An Analysis of Core Deformations in Protein Superfamilies},
+	journal = {Biophysical Journal},
+	volume = {88},
+	number = {2},
+	pages = {1291-1299},
+	year = {2005},
+	issn = {0006-3495},
+	doi = {https://doi.org/10.1529/biophysj.104.052449},
```
Comment (orbeckst) [inline package/doc/sphinx/source/references.bib]: just doi

File: package/doc/sphinx/source/references.bib [hunk 11]
```diff
@@ -734,4 +734,151 @@ @incollection{Waveren2005
...
+	volume = {185},
+	number = {6},
+	pages = {1546-1553},
+	year = {2014},
+	issn = {0010-4655},
+	doi = {https://doi.org/10.1016/j.cpc.2014.01.018},
+	author = {Pierre de Buyl and Peter H. Colberg and Felix Höfling},
+}
+
+@article{Jakupovic2021,
```
Comment (orbeckst) [inline package/doc/sphinx/source/references.bib]: This should be an "InProceedings" 
```bibtex

@inproceedings{Jakupovic:2021aa,
	address = {Austin, TX},
	author = {{E}dis {J}akupovic and {O}liver {B}eckstein},
	booktitle = {{P}roceedings of the 20th {P}ython in {S}cience {C}onference},
	editor = {{M}eghann {A}garwal and {C}hris {C}alloway and {D}illon {N}iederhut and {D}avid {S}hupe},
	pages = {40-48},
	title = {{MPI}-parallel {M}olecular {D}ynamics {T}rajectory {A}nalysis with the {H}5{M}{D} {F}ormat in the {MDAnalysis} {P}ython {P}ackage},
	year = {2021},
        doi = {10.25080/majora-1b6fd038-005}
}
```

**Review by orbeckst — CHANGES_REQUESTED** [review]
looking good, only a few easily fixable issues

File: package/doc/sphinx/source/documentation_pages/references.rst (hunk 3 above)
Comment (p-j-smith) [reply package/doc/sphinx/source/documentation_pages/references.rst:146]: Yes please!

File: package/doc/sphinx/source/documentation_pages/references.rst (hunk 7 above)
Comment (IAlibay) [reply package/doc/sphinx/source/documentation_pages/references.rst:36]: One paper at the time please 🙈

File: package/MDAnalysis/analysis/helix_analysis.py (hunk 5 above)
Comment (ojeda-e) [reply package/MDAnalysis/analysis/helix_analysis.py]: Oh, thanks for catching this, completely missed it.

**Comment by ojeda-e** [comment]
Thanks for all the comments @orbeckst, I'll address them in the next push, but also I think I may need some extra time here because the docs are not building properly and I can't identify the root of the issue. It is either (something I introduced in) [package/MDAnalysis/analysis/rms.py](https://github.com/MDAnalysis/mdanalysis/pull/3775/files#diff-71864c74fb263e02a38c28b542994b2e6ebae012f28fd35a57f5be291feff668R38), or [package/MDAnalysis/analysis/align.py](https://github.com/MDAnalysis/mdanalysis/pull/3775/files#diff-c33849e94200ad1054818e1f1f1633bd4a4972109e3808d1160e79896e3bcbb0R33), or [package/MDAnalysis/lib/qcprot.pyx](https://github.com/MDAnalysis/mdanalysis/pull/3775/files#diff-abff437607e28680e6a0afbf7e263f305c44c3e410d2f369e4e783613c856276R90). Something there is making docs break in the [references page](https://mdanalysis--3775.org.readthedocs.build/en/3775/documentation_pages/references.html) and similarly, in [align](https://mdanalysis--3775.org.readthedocs.build/en/3775/documentation_pages/analysis/align.html). 
Weirdly enough, docs build properly locally. I'll check what's going on.

File: package/MDAnalysis/analysis/helix_analysis.py (hunk 4 above)
Comment (ojeda-e) [reply package/MDAnalysis/analysis/helix_analysis.py]: I opted for adding the web archive link in both `references.rst` and `helix_analysis.py`. Better to fix this now while we are working on it.

**Comment by ojeda-e** [comment]
@orbeckst I addressed all your comments, thanks for pointing out the missing reference and the inconsistency of the dois in the bib references.

I found that updating references in `package/MDAnalysis/lib/qcprot.pyx` is problematic since it is breaking docs (see outdated [references page](https://mdanalysis--3775.org.readthedocs.build/en/3775/documentation_pages/references.html) ). It isn't clear to me why it fails here. Docs in `.pyx` files [is supported](https://github.com/sphinx-doc/sphinx/issues/1117). For now, I restored the previous version (old format for two references) to build docs properly. @MDAnalysis/docteam: Suggestions on how to update the refs to BibTex format without breaking docs are welcome.

Not sure if it's best to open a separate issue for the missing references in `qcprot.pyx` or try to fix it in this PR (with some extra help). If we choose the first one, this PR is ready.

**Comment by orbeckst** [comment]
Looks awful when it breaks :-p.

Did you successfully include bibtex citations from any other pyx files?

Did you try removing citations piece by piece to see if it's any specific one?

**Comment by ojeda-e** [comment]
> Looks awful when it breaks :-p.

Agreed!

> Did you successfully include bibtex citations from any other pyx files?

Not really. The citations I found in other pyx files are restricted to the header that reads "# Please cite your use of MDAnalysis in published work: (...)" , such as
-  [package/MDAnalysis/lib/_cutil.pyx](https://github.com/MDAnalysis/mdanalysis/blob/735abb8c76d71e1dc8197d73ee6f5d1845634914/package/MDAnalysis/lib/_cutil.pyx#L10)

- [package/MDAnalysis/lib/_augment.py](https://github.com/MDAnalysis/mdanalysis/blob/735abb8c76d71e1dc8197d73ee6f5d1845634914/package/MDAnalysis/lib/_augment.pyx), 

- or even [nsgrid.pyx](https://github.com/MDAnalysis/mdanalysis/blob/735abb8c76d71e1dc8197d73ee6f5d1845634914/package/MDAnalysis/lib/nsgrid.pyx) that mentions "Frenkel and Smit" but has no actual reference.

The only `.pyx` file with references to update was `qcprot.pyx` in [package/MDAnalysis/lib/qcprot.pyx](https://github.com/MDAnalysis/mdanalysis/blob/735abb8c76d71e1dc8197d73ee6f5d1845634914/package/MDAnalysis/lib/qcprot.pyx), references Theobald2005 and Liu2010.

> Did you try removing citations piece by piece to see if it's any specific one?

Yes, I did. Both Theobald2005 and Liu2010 in `qcprot.pyx`. I tried to add them separately, one by one. No matter which one I add, the docs break as soon as there is BibTex format on it.

**Comment by orbeckst** [comment]
Are there any intermediate files that are created? If so, do the references look correct in the intermediate files?

The issue about apidoc and pyx files https://github.com/sphinx-doc/sphinx/issues/1117 only addresses sphinx itself but does not mention bibtex. Is there any indication that the bibtex part might break??

**Comment by ojeda-e** [comment]
> Are there any intermediate files that are created? If so, do the references look correct in the intermediate files?

Not that I am aware of. The error shows a wrong indentation very far up from the line I edited. The reference was added in line 90 (see [commit](https://github.com/MDAnalysis/mdanalysis/pull/3775/commits/5541a8119b016b646978744768c779e74ee06b3f#diff-abff437607e28680e6a0afbf7e263f305c44c3e410d2f369e4e783613c856276R90)) while the error suggests (see [step 5 of the CI](https://github.com/MDAnalysis/mdanalysis/runs/7799175279?check_suite_focus=true#step:5:1374)) a wrong indentation in line 12-13:
```
docstring of MDAnalysis.lib.qcprot:12: ERROR: Unexpected indentation.
docstring of MDAnalysis.lib.qcprot:13: WARNING: Block quote ends without a blank line; unexpected unindent.
```
Please note that the error shows that the error is triggered when only one reference in BibTex is added. 

> The issue about apidoc and pyx files [sphinx-doc/sphinx#1117](https://github.com/sphinx-doc/sphinx/issues/1117) only addresses sphinx itself but does not mention bibtex. Is there any indication that the bibtex part might break??

Unfortunately, I am failing to find information on the topic. Since the first time docs break, I tried to dig in but all the info I find related to support is related to Sphinx only (as the one you pointed out), not Sphinx + sphinxcontrib-bibtex extension. 
I sent a message to the Sphinx mailing list asking for information (since it is a extension, not a problem on Sphinx itself). I also opened an issue in `sphinxcontrib-bibtex` asking if cython files are supported. See [sphinxcontrib-bibtex/issues#308](https://github.com/mcmtroffaes/sphinxcontrib-bibtex/issues/308).

**Comment by ojeda-e** [comment]
After checking that the BibTex extension is supported for Cython files, I tried several approaches to solve the problem but I failed to find a solution. From what I can see, the problem is in the Theobald reference, which happened to be in a Cython file as well. Adding the reference to the bibliography list breaks the HTML when adding it to `diffusionmap.py` or `references.rst`, and also happens when added in the cython file, `qcprot.pyx`. Honestly, I can't figure out what is the reason behind it despite trying several options. This is getting very complicated to debug because despite having identical versions for all sphinx packages and associated extensions, docutils, and other relevant packages, what I build locally is dramatically different from what is generated in the CI. 

Example: snapshots of `MDAnalysis.analysis.rms`
**local build**
image
vs. 
**CI build**
image

Additionally, building locally doesn't trigger errors than are caught by the CI.

I am kind of giving up here because no matter what I try the docs break. I tried doing a diff between the in `rms.py` HTML files generated locally and the CI and the output is more than I'd expect. Here is a snapshot, where the one on the left is generated by the CI and the one on the right is generated locally.
image
The one from the CI is also unexpected in the version of docutils. I would expect docutils==0.17.1, as shown in during [building](https://github.com/MDAnalysis/mdanalysis/runs/8059260447?check_suite_focus=true#step:5:64), not 0.18.1, as seems to be the one used to build docs (BTW, if someone can explain to me why is there a difference, I'd appreciate it).

In summary, I don't know how to proceed here. I'm tempted to put Theobald references back to rst and have something functional that doesn't break the three problematic pages: `package/MDAnalysis/analysis/rms.py` `package/MDAnalysis/analysis/align.py`, `qcprot.pyx`and `references.rst`.

Any suggestions are more than welcome.

**Comment by orbeckst** [comment]
Thank you for digging deeply into the bibtex extension and checking with upstream. All of this is really important when we adopt a new technology — we want to understand problems at the beginning.

Just so that I understand what's happening — please correct me:
1. You narrowed down problems to the Theobald reference that is used in a pyx file as well as other files. If you don't have it in the pyx then no problems appear.
2. Building the docs locally works and they appear as expected.
3. The builds that are displayed on RTD as part of the PR build differ visually from the local ones.
4. "Additionally, building locally doesn't trigger errors than are caught by the CI." — what are the errors that you refer to? Did you mean the "docstring of MDAnalysis.lib.qcprot:12: ERROR: Unexpected indentation."?

Concerning (3), what is the difference that you consider incorrect?

(I am asking because as far as I can tell, RTD doc formatting always differs slightly between our published docs and the RTD version, see for example [rms module in PR 3799](https://mdanalysis--3799.org.readthedocs.build/en/3799/documentation_pages/analysis/rms.html) vs [rms module docs v2.2](https://docs.mdanalysis.org/stable/documentation_pages/analysis/rms.html).)

About (4): Having sphinx throw errors is annoying as it stops tests. Finding a fix for it would be important. Perhaps adding additional text afterwards, checking spaces vs tabs, ... and other Voodo-coding might help? (... it's super-annoying, I know ...)

Perhaps the fact that the RTD html has `class="no-js"` and ours has `class="writer-html5"` is significant. I don't know. However, as long as the RTD version for your PR looks similar to the RTD version on other PRs _and_ you confirmed locally that it builds correctly _and_ sphinx does not throw errors, then we can chance it and make the changes and see what it looks like in production devel docs.

Maybe @lilyminium has some more insights?

**Comment by orbeckst** [comment]
@ojeda-e did I understand your findings correctly? 

Any idea what else to try or how to solve this problem, maybe with a work-around?

**Comment by ojeda-e** [comment]
Sorry for the late reply @orbeckst. For some reason, Github hasn't sent me activity notifications for a while. 
> Just so that I understand what's happening — please correct me:
> 1. You narrowed down problems to the Theobald reference that is used in a pyx file as well as other files. If you don't > have it in the pyx then no problems appear.

Correct.
> 2. Building the docs locally works and they appear as expected.
> 3. The builds that are displayed on RTD as part of the PR build differ visually from the local ones.

Correct.

> 4. "Additionally, building locally doesn't trigger errors than are caught by the CI." — what are the errors that you refer to? Did you mean the "docstring of MDAnalysis.lib.qcprot:12: ERROR: Unexpected indentation."?

Yes, exactly that.

> (I am asking because as far as I can tell, RTD doc formatting always differs slightly between our published docs and the RTD version, see for example [rms module in PR 3799](https://mdanalysis--3799.org.readthedocs.build/en/3799/documentation_pages/analysis/rms.html) vs [rms module docs v2.2](https://docs.mdanalysis.org/stable/documentation_pages/analysis/rms.html).)

Oh. I didn't know that was a thing. I really appreciate you pointing that out. Was debugging step by step to find out why the rendered docs looked like that. The same happens in the docs here updated, so it won't probably be an issue.

If there is a chance that the docs may not render properly, I could offer a solution. Considering we have Theobald2005 in several files:

- `./doc/sphinx/source/documentation_pages/references.rst`
- `./MDAnalysis/analysis/align.py`
- `./MDAnalysis/analysis/diffusionmap.py`
- `./MDAnalysis/analysis/rms.py`
- `./MDAnalysis/lib/qcprot.pyx` 

Locally, the docs render as expected using the following prefixes for different files:

|     **file**    	| **prefix** 	|
|:---------------:	|:----------:	|
|  references.rst 	|      ᵃ     	|
|     align.py    	|      ᵇ     	|
| diffusionmap.py 	|      -     	|
|      rms.py     	|      ᵈ     	|
|    qcprot.pyx   	|      ᶜ     	|

(Other than the one for references.rst `ᵃ`, because I wanted consistency on that page, the prefixes were assigned randomly; `-` means no prefix, equivalent to an empty prefix.)

With this solution, the `references.rst` page, where Theobald shows up would look like this (rendered locally):
image

This solution renders locally for all the other files above listed. The only thing is that this will likely look different on the CI.

Would it be too risky or complicated to create something like a test issue where I only change Theobald, merge it, and check how does it look like and if it's too bad we undo the merge?

**Comment by orbeckst** [comment]
@ojeda-e my turn to apologize... sorry, lost track of the PR. 

If you have a solution that renders the reference and does not crash CI/RTD then I am happy with it. 

I looked at https://mdanalysis--3775.org.readthedocs.build/en/3775/documentation_pages/references.html#citations-for-included-algorithms-and-modules and the list of references for included algorithms looks like a mess, as you indicated. If it _only_ looks bad on RTD but ok in our docs https://docs.mdanalysis.org then I'd be ok with it. I don't know if e.g. @lilyminium has a different opinion.

> Would it be too risky or complicated to create something like a test issue where I only change Theobald, merge it, and check how does it look like and if it's too bad we undo the merge?

Do you want to do a single change and then see how it shows up at https://docs.mdanalysis.org to avoid looking at it through the lens of RTD?

If yes, then please go ahead and create that PR. If the other one does what we expect (looks ok in our docs, broken references page on RTD) then we can merge develop into this PR and ultimately merge this one, too.

**Comment by lilyminium** [comment]
> docstring of MDAnalysis.lib.qcprot:12: ERROR: Unexpected indentation

Hmm, yes, I'd say go ahead and see the change works but maybe raise a new (low-priority) issue to check out what's going on with CI... maybe the docs dependency versions need updating or something similar.

If you want to do a dry-run on a fork, you can change CI to build to a github-pages URL of your own. I did that here: https://mdadocs.minium.com.au/2.4.0-dev0/documentation_pages/references.html#citations-for-included-algorithms-and-modules is current for commit [79d4feba2dbc0adbbe2bfd21cd0bfa33892510ce](https://github.com/MDAnalysis/mdanalysis/commit/79d4feba2dbc0adbbe2bfd21cd0bfa33892510ce)

(it's the full docs so you can click around, but every now-and-then you might hit a redirect page that sends you to docs.mdanalysis.org)

**Comment by hmacdope** [comment]
There was some discussion of the need for `pyyaml` in #3842 https://github.com/MDAnalysis/mdanalysis/pull/3842/files#r1027222362

Not sure if relevant.

**Comment by IAlibay** [comment]
@ojeda-e @orbeckst is this still PR still active?

**Comment by orbeckst** [comment]
In principle, these changes still need to be done to change references, as far as I can tell.

I merged develop into this branch and maybe citations become magically correct?? (Hope springs eternal...)

**Comment by orbeckst** [comment]
@lilyminium from the conflicts I see that `footblibliography` appears in place of the explicit footnotes from this PR. Does this mean that this PR is no longer needed?

**Comment by lilyminium** [comment]
Yes and no -- @ojeda-e put a lot of amazing work into fixing up all our references and making sure everything was neatly bibtex that would be a great pity to lose, and I only went through and fixed existing, previously merged bibliographies when I updated the theme. Presumably the rest of the 18 files with changes that don't have conflicts are all improvements of changing footnotes to bibtex references. I'm not sure how much @ojeda-e wants to keep working on this, but it's been quite a while -- maybe the easiest way to resolve this PR would be for me to either take over and resolve the conflicts/convert things into local bibliographies, or cherry-pick commits into a different PR and go from there.

**Comment by ojeda-e** [comment]
Hi @lilyminium, please feel free to cherry pick anything is useful for you and close this PR if that's the best approach for the organization to move forward. Thanks for acknowledging the work, I appreciate it.
